app/models/summaryReturnsModels.scala

```diff
@@ -98,11 +100,32 @@ object SummaryReturnsModel {
       val atedBalance: Option[BigDecimal] = (json \ "atedBalance").asOpt[BigDecimal]
       val allReturns: Seq[PeriodSummaryReturns] = (json \ "allReturns").as[Seq[PeriodSummaryReturns]]
 
-      val returnsCurrentTaxYear: Seq[PeriodSummaryReturns] = allReturns.filter(
+      val allReturnsForSocialHousing: Seq[PeriodSummaryReturns] = allReturns.map { returnsForPeriod =>
+        if (returnsForPeriod.periodKey >= 2020 && applicationConfig.isEnabled(FeatureSwitch.CooperativeHousing)) {
+          returnsForPeriod.copy(submittedReturns = returnsForPeriod.submittedReturns.map( submittedReturns =>
+            submittedReturns.copy(reliefReturns = submittedReturns.reliefReturns.map( reliefReturn =>
+              if (reliefReturn.reliefType == "Social housing") {
+                reliefReturn.copy(reliefType = newSocialHousingDescription)
+              } else {
+                reliefReturn
+              }
+            )
+          )), draftReturns = returnsForPeriod.draftReturns.map {
+            case draftReturn @ DraftReturns(periodKey, _, "Social housing", _, "Relief") if periodKey >= 2020 =>
+              draftReturn.copy(description = newSocialHousingDescription)
+            case draftReturn =>
```

Comment:
Does this need an underscore case?

Reply:
It is one, it's just bound to a variable